src/extension-support/extension-manager.js

```diff
@@ -375,29 +375,36 @@ class ExtensionManager {
             blockAllThreads: false,
             arguments: {}
         }, blockInfo);
-        blockInfo.opcode = this._sanitizeID(blockInfo.opcode);
+        blockInfo.opcode = blockInfo.opcode && this._sanitizeID(blockInfo.opcode);
         blockInfo.text = blockInfo.text || blockInfo.opcode;
 
-        if (blockInfo.blockType !== BlockType.EVENT) {
+        switch (blockInfo.blockType) {
+        case BlockType.EVENT:
+            if (blockInfo.func) {
+                log.warn(`Ignoring function "${blockInfo.func}" for event block ${blockInfo.opcode}`);
+            }
+            break;
+        case BlockType.BUTTON:
+            if (blockInfo.opcode) {
+                log.warn(`Ignoring opcode "${blockInfo.opcode}" for button with text: ${blockInfo.text}`);
+            }
+            break;
+        default:
             blockInfo.func = blockInfo.func ? this._sanitizeID(blockInfo.func) : blockInfo.opcode;
```

Comment:
Should we log a warning or an error if opcode is undefined here? Or if both opcode and func are undefined? Alternatively, if they are undefined, should we default to a built in no-op function?

Reply:
Ah, never mind, I see we throw an error below. Nevertheless, I'll leave the question about defaulting to a no-op function here.

Reply:
Good idea -- done. Note that the exception thrown here will be caught and reported by some code in `_prepareExtensionInfo` which dumps the whole `blockInfo` so there's no need to add any `blockInfo` fields in the exception message.